xla/service/gpu/runtime/thunk.h

```diff
@@ -270,6 +270,10 @@ class Thunk {
     int64_t collective_max_nchannels;
     int64_t p2p_max_nchannels;
 
+    absl::Status async_status = absl::OkStatus();
```

Comment:
nit: can you add a comment explaining the purpose of these two?

Reply:
Yes sure. Also I'm moving these to gpu_executable since it makes more sense for these to be part of the global executable. I should have another commit by EOD. @frgossen can we hold off on merging it today? i will temporarily change it to draft for now.

Reply:
moved the async queue and async status to gpu executable. please take another look. thanks.